tests/filecheck/dialects/emitc/emitc_types_invalid_parsing.mlir

```diff
@@ -6,3 +6,10 @@
 "test.op"() {
   empty = !emitc.array<f32>
 }: ()->()
+
+// -----
+
+// CHECK: xdsl.utils.exceptions.VerifyException: EmitC array element type 'i0' is not a supported EmitC type.
```

Comment:
I'm not sure how to catch the `VerifyException` here

Reply:
This is why I was suggesting to first open a PR to remove the parsing diagnostic test, let's open that one to migrate this test to to catch verification errors instead.